
Config is: GlobalConfig(image=ImageConfig(image_size=28, patch_size=7, n_channels=1), transformer=TransformerConfig(hidden_dim=128, num_heads=4, num_layers=4, mlp_dim=512, activation_name='GELU', attention_only=True, attn_hidden_layer=True), layernorm=LayerNormConfig(qknorm=False, layer_norm_eps=0.0), dropout=DropoutConfig(patch=0.0, position=0.0, attention=0.0, proj=0.0, mlp=0.0), init=InitializationConfig(weight_type='he', cls_std=1e-06, pos_std=0.02), training=TrainingConfig(loss_fn_name='CrossEntropy', lr=0.001, num_epochs=5, batch_size=64, warmup_steps=0, weight_decay=0.0, device='cuda', seed=0, optimizer_name='AdamW'), logging=LoggingConfig(log_dir='logs', log_frequency=10, print_every=0, use_wandb=True), saving=SavingConfig(parent_dir='/network/scratch/s/sonia.joseph/vit_planetarium', save_dir='checkpoints', save_checkpoints=True, save_cp_frequency=10), classification=<vit_planetarium.configs.MNISTConfig.ClassificationConfig object at 0x7f50b9827450>)
  0%|                                                                                                                                                                                                                                                                           | 0/5 [00:00<?, ?it/s]

0it [00:05, ?it/s]

9it [00:06,  1.91it/s]

21it [00:09,  3.07it/s]
Steps20 | Train loss: 0.066 | Train acc: 0.285 | Test loss: 0.067 | Test acc: 0.287

33it [00:11,  4.27it/s]

50it [00:12,  7.32it/s]

54it [00:14,  5.48it/s]
Steps60 | Train loss: 0.023 | Train acc: 0.529 | Test loss: 0.025 | Test acc: 0.486

74it [00:17,  5.81it/s]

89it [00:18,  7.93it/s]
  0%|                                                                                                                                                                                                                                                                           | 0/5 [00:21<?, ?it/s]
Traceback (most recent call last):
  File "test_trainer.py", line 29, in <module>
    unittest.main()
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/main.py", line 101, in __init__
    self.runTests()
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/main.py", line 271, in runTests
    self.result = testRunner.run(self.test)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/runner.py", line 176, in run
    test(result)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/suite.py", line 122, in run
    test(result)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/suite.py", line 84, in __call__
    return self.run(*args, **kwds)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/suite.py", line 122, in run
    test(result)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/case.py", line 676, in __call__
    return self.run(*args, **kwds)
  File "/cvmfs/ai.mila.quebec/apps/x86_64/debian/anaconda/3/lib/python3.7/unittest/case.py", line 628, in run
    testMethod()
  File "test_trainer.py", line 25, in test_train_function
    trained_model = train(self.model, self.config, self.train_dataset, self.val_dataset)
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/src/vit_planetarium/training/trainer.py", line 60, in train
    test_loss = calculate_loss(model, test_loader, loss_fn, config.training.device)
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/src/vit_planetarium/training/training_utils.py", line 27, in calculate_loss
    for items in islice(data_loader, N // batch_size):
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 628, in __next__
    data = self._next_data()
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/torch/utils/data/dataloader.py", line 671, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 58, in fetch
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/torch/utils/data/_utils/fetch.py", line 58, in <listcomp>
    data = [self.dataset[idx] for idx in possibly_batched_index]
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/torchvision/datasets/mnist.py", line 142, in __getitem__
    img = Image.fromarray(img.numpy(), mode="L")
  File "/home/mila/s/sonia.joseph/ViT-Planetarium/env/lib/python3.7/site-packages/PIL/Image.py", line 3081, in fromarray
    strides = arr.get("strides", None)
KeyboardInterrupt